extension/httpfs/s3fs.cpp

```diff
@@ -154,7 +154,8 @@ void AWSEnvironmentCredentialsProvider::SetExtensionOptionValue(string key, cons
 }
 
 void AWSEnvironmentCredentialsProvider::SetAll() {
-	this->SetExtensionOptionValue("s3_region", this->REGION_ENV_VAR);
+	const char *region_env_var = this->REGION_ENV_VAR != NULL ? this->REGION_ENV_VAR : this->DEFAULT_REGION_ENV_VAR;
```

Comment:
This isn't quite correct - you probably want to instead call SetExtensionOptionValue twice instead

Reply:
For some reason I had it in my head that i needed the conditional, as if AWS_REGION wasn't set it'd override AWS_DEFAULT_REGION with null - but I completely neglected that it checks if the environment variable is empty in `SetExtensionOptionValue`.

Have updated now, sorry about that.